rust/src/syscore.rs

```diff
@@ -0,0 +1,82 @@
+pub use self::ffi::*;
+use std::ptr::NonNull;
+
+/// Opaque type for C interop: RpmOstreeOrigin.
+pub enum RpmOstreeOrigin {}
+
+extern "C" {
+    fn rpmostree_origin_get_live_state(
```

Comment:
I'm OK with this but we need to be careful because this would mean changing that API C side in the future would result in crashes/corruption on this side.

Let's at least add a comment to both sides warning to change the other; something like this in C:
`// WARNING: This prototype is also redefined in Rust, if changing this please also update the Rust side`
And here:
`// NOTICE: The C header definition is canonical, please update that first then synchronize this`

Reply:
Ack.
There is another one already bridged (`rpmostree_get_repodata_chksum_repr`), do you prefer to have all of them in a single place/module in a Rust and a single top-notice?